File: src/main/java/dev/zprestige/fire/module/movement/speed/KeyListener.java
package dev.zprestige.fire.module.movement.speed;

import dev.zprestige.fire.event.bus.EventListener;
import dev.zprestige.fire.event.impl.KeyEvent;

public class KeyListener extends EventListener<KeyEvent, Speed> {

    public KeyListener(final Speed speed) {
        super(KeyEvent.class, speed);
    }

    @Override
    public void invoke(final Object object) {
        final KeyEvent event = (KeyEvent) object;
        if (module.switchKey.GetKey() == event.getKey()) {
            if (module.speedMode.GetCombo().equals("Strafe")) {
                module.speedMode.setValue("OnGround");
            } else {
                module.speedMode.setValue("Strafe");
            }
            module.sendSwitchMessage();
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/movement/speed/KeyListener.kt
package dev.zprestige.fire.module.movement.speed

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.KeyEvent

class KeyListener(speed: Speed) : EventListener<KeyEvent, Speed>(KeyEvent::class.java, speed) {
    
    override fun invoke(e: Any) {
        val event = e as KeyEvent
        if (module.switchKey.GetKey() == event.key) {
            if (module.speedMode.GetCombo() == "Strafe") {
                module.speedMode.value = "OnGround"
            } else {
                module.speedMode.value = "Strafe"
            }
            module.sendSwitchMessage()
        }
    }
}